src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -15,6 +15,11 @@ public class ToolSettings
         public FilePath ToolPath { get; set; }
 
         /// <summary>
+        /// Gets or sets optional timeout for tool execution
+        /// </summary>
+        public int? ToolTimeout { get; set; }
```

Comment:
Hmm don't totally agree here. Happy do add InMilliseconds suffix.
Wouldn't just Timeout potentially collide with any setting that inherits? Think just Timeout is too generic and as it's two words I think we should case it ToolTimeoutInMilliseconds.